vendor/k8s.io/kubernetes/staging/src/k8s.io/apiserver/pkg/endpoints/installer.go

```diff
@@ -589,15 +589,6 @@ func (a *APIInstaller) registerResourceHandlers(path string, storage rest.Storag
 
 		routes := []*restful.RouteBuilder{}
 
-		// If there is a subresource, kind should be the parent's kind.
-		if hasSubresource {
```

Comment:
I don't think we should do this.  This is messing with the kinds reported via discovery, right?